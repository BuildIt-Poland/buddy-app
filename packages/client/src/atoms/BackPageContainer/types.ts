export interface BackPageContainerProps {
  title?: string;
  id?: string;
  children?: any;
  backGroundShape?: boolean;
  navItems?: React.ReactNode;
}
