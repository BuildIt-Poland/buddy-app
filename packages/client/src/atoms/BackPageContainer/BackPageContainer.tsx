import React from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import { goBack } from 'utils';
import Header, { MenuTypes } from 'components/Header';
import PageContainer from 'atoms/PageContainer';
import { BackPageContainerProps } from './types';

const useStyles = makeStyles(theme => ({
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
  },
  header: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    textTransform: 'capitalize',
    margin: theme.spacing(1, 0, 3),
  },
}));

const BackPageContainer: React.FC<BackPageContainerProps> = ({
  id,
  title,
  backGroundShape,
  navItems,
  children,
}) => {
  const { wrapper, header } = useStyles();
  const history = useHistory();
  const onButtonClick = () => goBack(history);

  return (
    <>
      <Header
        type={MenuTypes.BACK}
        onButtonClick={onButtonClick}
        navItems={navItems}
      />
      <PageContainer backGroundShape={backGroundShape} data-testid={id}>
        <Box className={wrapper}>
          {title && (
            <Box className={header}>
              <Typography component='h2' variant='h2'>
                {title}
              </Typography>
            </Box>
          )}
          {children}
        </Box>
      </PageContainer>
    </>
  );
};

export default BackPageContainer;
