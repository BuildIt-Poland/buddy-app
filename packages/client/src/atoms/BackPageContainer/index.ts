import BackPageContainer from './BackPageContainer';

export default BackPageContainer;
