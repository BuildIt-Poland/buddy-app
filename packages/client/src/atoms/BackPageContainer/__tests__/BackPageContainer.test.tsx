import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter, Route } from 'react-router';
import BackPageContainer from '../index';

jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('atoms/PageContainer', () => 'PageContainer');
jest.doMock('components/Header');

describe('Component - BackPageContainer', () => {
  it('renders correctly', async () => {
    const path = '/buddy/newbies/1234/details';
    const component = create(
      <MemoryRouter initialEntries={[path]}>
        <Route path={'/buddy/newbies/:newbieId/details'}>
          <BackPageContainer>hello world</BackPageContainer>
        </Route>
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
