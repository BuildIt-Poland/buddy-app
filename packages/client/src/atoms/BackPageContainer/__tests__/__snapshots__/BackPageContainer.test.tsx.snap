// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - BackPageContainer renders correctly 1`] = `
Array [
  <header
    type="back"
  />,
  <PageContainer>
    <Box
      className="makeStyles-wrapper-1"
    >
      hello world
    </Box>
  </PageContainer>,
]
`;
